File: src/gui/itemnode/textfield/ArrowDialogButton.java
package gui.itemnode.textfield;

import de.jensd.fx.glyphs.fontawesome.FontAwesomeIcon;
import gui.objects.icon.Icon;
import javafx.scene.layout.StackPane;

/**
 * Button for calling dialogs, from within {@link javafx.scene.control.TextField}.
 */
public class ArrowDialogButton extends StackPane {

	public ArrowDialogButton() {
		// Non-icon pure css implementation, that looks exactly like other javaFx dialog
		// String styleClass = "dialog-button";
		// Region r = new Region();
		//        r.getStyleClass().add(styleClass);
		//        r.setMinSize(0, 0);
		//        r.setPrefSize(7, 6);
		//        r.setMaxSize(7, 6);
		// setPrefSize(22,22);

		Icon r = new Icon(FontAwesomeIcon.CARET_DOWN, 7).scale(2).tooltip("Open dialog");
		getChildren().add(r);
	}
}

File: src/sp/it/pl/gui/itemnode/textfield/ArrowDialogButton.kt
package sp.it.pl.gui.itemnode.textfield

import javafx.scene.layout.StackPane
import sp.it.pl.gui.objects.icon.Icon

/** Button for calling dialogs, from within [javafx.scene.control.TextField]. */
class ArrowDialogButton: StackPane() {
    init {
        // Non-icon pure css implementation, that looks exactly like other javaFx dialog
        // children +=  Region().apply {
        //     styleClass += "dialog-button"
        //     styleClass += "value-text-field-arrow"
        //     setMinSize(0.0, 0.0)
        //     setPrefSize(7.0, 6.0)
        //     setMaxSize(7.0, 6.0)
        // }
        // setPrefSize(22.0, 22.0)

        children += Icon().tooltip("Open dialog").styleclass("value-text-field-arrow")
    }
}